src/coreComponents/codingUtilities/Utilities.hpp

```diff
@@ -414,6 +415,51 @@ struct BitFlags
 
 };
 
+template< typename VECTOR_TYPE, typename SCALAR_TYPE >
+VECTOR_TYPE addTwoVecs( VECTOR_TYPE const & vec1,
+                        VECTOR_TYPE const & vec2,
+                        SCALAR_TYPE const sign )
+{
+  GEOS_ASSERT( vec1.size() == vec2.size() );
+  const localIndex N = vec1.size();
+  VECTOR_TYPE result( N );
+  RAJA::forall< parallelHostPolicy >( RAJA::TypedRangeSegment< localIndex >( 0, N ),
+                                      [&] GEOS_HOST_DEVICE ( localIndex const i )
+  {
+    result[i] = vec1[i] + sign * vec2[i];
+  } );
+  return result;
+}
+
+template< typename VECTOR_TYPE, typename SCALAR_TYPE >
+VECTOR_TYPE scalarMultiplyAVec( VECTOR_TYPE const & vec,
+                                SCALAR_TYPE const scalarMult )
+{
+  const localIndex N = vec.size();
+  VECTOR_TYPE result( N );
+  RAJA::forall< parallelHostPolicy >( RAJA::TypedRangeSegment< localIndex >( 0, N ),
+                                      [&] GEOS_HOST_DEVICE ( localIndex const i )
+  {
+    result[i] = scalarMult * vec[i];
+  } );
+  return result;
+}
+
+template< typename VECTOR_TYPE >
+real64 dotTwoVecs( VECTOR_TYPE const & vec1,
+                   VECTOR_TYPE const & vec2 )
+{
+  GEOS_ASSERT( vec1.size() == vec2.size());
+  RAJA::ReduceSum< parallelHostReduce, real64 > result( 0.0 );
+  const localIndex N = vec1.size();
+  RAJA::forall< parallelHostPolicy >( RAJA::TypedRangeSegment< localIndex >( 0, N ),
+                                      [&] GEOS_HOST_DEVICE ( localIndex const i )
+  {
+    result += vec1[i] * vec2[i];
+  } );
+  return result.get();
+}
+
```

Comment:
do you want to see the linear algebra vectors use in this PR or it's ok to do it later in another PR?

Reply:
No need to have linear algebra vectors use in this PR. Just wanted to make sure that's the plan to remove the unit test requirement.

Reply:
@francoishamon pointed to Utilities (https://github.com/GEOS-DEV/GEOS/pull/2628#discussion_r1358345287)
what would be a better place?

Reply:
> I would do it directly in this PR because I am afraid that otherwise it won't happen.

If we open an issue, we won't forget.

> I am also not sure Utilities is the right location for these functions.

This is a linear algebra capability, so the natural place would be: `src/coreComponents/linearAlgebra/utilities`. However, it is a new implementation of methods already provided by the LAI.

Reply:
create issue here https://github.com/GEOS-DEV/GEOS/issues/2891
just let me know where to place those functions, is src/coreComponents/linearAlgebra/utilities ok or not?

Reply:
Honestly, those functions will disappear. Let's keep them there for now, no need to look for a fancy temporary solution.